File: examples/kotlin/features/boundary-protocols/kotlin/cott_impl/curriculum/boundary_protocols/wrap_handle.kt
package cott_impl.curriculum.boundary_protocols

internal fun wrap_handle(raw_id: kotlin.ULong): cott_runtime.CottResult<curriculum.boundary_protocols.HandleBundle, curriculum.boundary_protocols.HandleError> {
    if (raw_id == 0uL) {
        return cott_runtime.Err(curriculum.boundary_protocols.HandleError.InvalidHandle)
    }
    return cott_runtime.Ok(
        curriculum.boundary_protocols.HandleBundle(
            handle = cott_runtime.Opaque.of(cott_runtime.CottOpaque_7237d55bbcbee604a7ba6463, raw_id),
            raw_id = raw_id
        )
    )
}

File: examples/features/boundary-protocols/python/_cott_impl/curriculum/boundary_protocols/wrap_handle.py
from typing import Literal

from cott_runtime import Opaque
from curriculum.boundary_protocols_types import ConnectionId, HandleBundle


def wrap_handle(raw_id: ConnectionId) -> HandleBundle:
    handle: Opaque[Literal["client_session"]] = Opaque(tag="client_session", value=raw_id.value)
    return HandleBundle(handle=handle, raw_id=raw_id)
